### ISSUE TITLE
How disconnect properly

### ISSUE THREAD

**Issue by spvince** [GetStream/stream-chat-android#524]
<h2>What did you do?</h2>
I don't know if it's a real issue or mistake of me.

1. I logged with user 28686.
2. I disconnect this user.
3. I log with user 21715

<h2>What did you expect to happen?</h2>
No crash

<h2>What happened instead?</h2>
I have a crash

<br/> <details>
<summary>Crash </summary>
<p>

```java
2020-06-11 18:13:00.839 3811-3962/io.studentpop.job.dev E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-8
    Process: io.studentpop.job.dev, PID: 3811
    java.util.InputMismatchException: received connect event for user with id 28686 while chat domain is configured for user with id 21715. create a new chatdomain when connecting a different user.
        at io.getstream.chat.android.livedata.ChatDomainImpl.updateCurrentUser$livedata_release(ChatDomainImpl.kt:193)
        at io.getstream.chat.android.livedata.EventHandlerImpl.updateOfflineStorageFromEvents$livedata_release(EventHandlerImpl.kt:225)
        at io.getstream.chat.android.livedata.EventHandlerImpl$updateOfflineStorageFromEvents$1.invokeSuspend(Unknown Source:12)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```
</p>
</details>

<br/> <details>
<summary>My method to disconnect streamChat </summary>

```java
    suspend fun disconnect() {
        Timber.d("disconnect")

        chatDomain.disconnect()
        chatDomain.clean()

        Chat.getInstance().client.disconnectSocket()
        Chat.getInstance().client.disconnect()
    }

```
</details>

<br/> <details>
<summary>My method when i log a new user </summary>

```java
    fun configureStreamClient() {
        Timber.d("configureStreamClient")

        val spUser = StudentSession.getCurrentUser()
        val userId = spUser?.id.toString()

        val userToken: String = spUser?.streamChatToken ?: ""

        val userName: String = spUser?.firstName ?: ""
        val userPicture: String = spUser?.pictureUrl ?: ""

        val extraData = HashMap<String, Any>()
        extraData[EXTRA_DATA_NAME] = userName
        extraData[EXTRA_DATA_PICTURE] = userPicture

        val user = User(userId)
        user.extraData = extraData

        chatDomain = ChatDomain.Builder(mContext, Chat.getInstance().client, user).build()

        Chat.getInstance().client.setUser(user, userToken, object : InitConnectionListener() {
            override fun onSuccess(data: ConnectionData) {
                super.onSuccess(data)

                Timber.d("configureStreamClient - success")
            }

            override fun onError(error: ChatError) {
                super.onError(error)

                Timber.e("configureStreamClient - error: $error")
            }
        })
    }
```
</details>

<h2>GetStream Environment</h2>

GetStream Chat version:
4.2.11-beta-4

**Comment by abhirup-patra** [comment]
changing user crashes the app. Is there any approximate fix date? Sometimes chat crashes when trying to fetch live data

**Comment by tbarbugli** [comment]
@samiuelson do we wait for the websocket connection to be properly clean when we call `disconnect`? If not we should and make sure you can wait for that to be completed *or* force close the connection without waiting for the server close reply.

@samiuelson we should also update our docs to mention chatDomain (logout only shows the LLC way), do you think we can return a more informative error message? Sounds like an easy configuration mistake to make

**Comment by abhirup-patra** [comment]
the continuation and subroutines are not part of jvm. Its a part of kotlin. i am not able to implement this in java

**Comment by JcMinarro** [comment]
Hello @spvince 
Could you check if it is fixed by #563
It is already part of the version 4.2.11-beta-6, not a final release yet, we will update #554 when the final version is ready

**Comment by aengussong** [comment]
Hi!
I can still reproduce this issue in the version 4.4.1. after disconnecting client and domain and then resetting the user and creating new ChatDomain `InputMismatchException` occurs
